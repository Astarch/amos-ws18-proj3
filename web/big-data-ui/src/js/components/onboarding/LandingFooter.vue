<template>
  <!-- Footer -->
  <footer>
    <div class="container">
      <div class="row">
        <div class="col-lg-12">
          <ul class="list-inline">
            <li>
              <a href="#">Home</a>
            </li>
            <li class="footer-menu-divider">&sdot;</li>
            <li>
              <a href="#about">About</a>
            </li>
            <li class="footer-menu-divider">&sdot;</li>
            <li>
              <a href="#services">Services</a>
            </li>
            <li class="footer-menu-divider">&sdot;</li>
            <li>
              <a href="#contact">Contact</a>
            </li>
          </ul>
          <p class="copyright text-muted small">Copyright &copy; pretrendr 2017. All Rights Reserved</p>
        </div>
      </div>
    </div>
  </footer>
</template>

<script>
  export default {
    name: 'landing-footer'
  }


</script>

<style lang="scss" scoped>

  footer {
    padding: 50px 0;
    background-color: #f8f8f8;
  }

  p.copyright {
    margin: 15px 0 0;
  }


</style>
