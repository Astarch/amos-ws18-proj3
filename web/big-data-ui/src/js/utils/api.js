import axios from 'axios';
import qs from 'qs';
import store from 'store2'


/******************** Base API / Client Configuration ********************/

let port = "8081/"
let localhostUrl = "http://localhost:" + port;
let stagingUrl = "http://18.216.129.153:" + port;
let liveUrl = "http://18.216.122.218:" + port;
let baseUrl = stagingUrl
baseUrl = (GIT.BRANCH && !GIT.BRANCH.includes("master")) ? liveUrl : stagingUrl;



function writeAuthToken(token) {
    store.session('authToken', token)
    return token
}

function getAuthToken() {
    return store.session('authToken')
}


const axiosInstance = axios.create({
    baseURL: baseUrl,
});

if (getAuthToken() && getAuthToken().length > 5) {
    Object.assign(axiosInstance.defaults, { headers: { 'x-auth-token': getAuthToken() } });
}

// Intercepts all responses, retrieves the x auth token and sets it to all following requests headers!
axiosInstance.interceptors.response.use(response => {
    let authToken = getAuthToken()
    if (response.status == 401) {
        authToken = writeAuthToken("")
    } else if (response && response.headers && response.headers['x-auth-token']) {
        let authToken = writeAuthToken(response.headers['x-auth-token']);
        if (authToken && authToken.length > 5) {
            Object.assign(axiosInstance.defaults, { headers: { 'x-auth-token': authToken } });
        }
    }

    return response;
}, error => {
    // Do something with response error
    return Promise.reject(error);
});

export default axiosInstance;

export const isLoggedIn = () => {
    let authToken = getAuthToken() ? getAuthToken() : authToken;
    return authToken && authToken.length > 5
}

export const logoutUser = () => {
    writeAuthToken("")
}



/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => axiosInstance.post('/auth/login', qs.stringify({
    username: username,
    password: password
}));

const postCompleteRegistration = (username,
    password,
    firstname = username,
    lastname = username,
    email = username + "@test.com",
    address = "addres1",
    phone = "phoneNr") => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "firstname": firstname,
    "lastname": lastname,
    "email": email,
    "address": address,
    "phone": phone
});

const postRegistration = (username,
    email,
    password) => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "email": email,
});

const doLogout = () => axiosInstance.get('/auth/logout');

const auth = {
    postLogin,
    postRegistration,
    postCompleteRegistration,
    doLogout
};

/***** user *****/
const getAll = () => axiosInstance.get('/api/user/getAll');

const user = {
    getAll
};

/***** graph *****/
const getData = (path) => axiosInstance.get(path);

const graph = {
    getData
};

const api = {
    auth,
    user,
    graph
}

export {
    api
};