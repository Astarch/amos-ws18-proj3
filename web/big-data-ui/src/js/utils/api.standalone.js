import axios from 'axios';
import qs from 'qs';
import storage from 'store2'
import store from 'src/js/store';

import trump from 'src/assets/wordcounts/trump'
const availableTerms = ["trump"]



/******************** Base API / Client Configuration ********************/

function writeAuthToken(token) {
  storage.set('authToken', token)
  return token
}

function getAuthToken() {
  return storage.get('authToken')
}

export const isLoggedIn = () => {
  let authToken = getAuthToken() ? getAuthToken() : authToken;
  return authToken && authToken.length > 5
}

export const logoutUser = () => {
  writeAuthToken("")
}


/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => {
  return new Promise((resolve, reject) => {
    if (username == "user2" && password == "pass2") {
      const response = {
        data: {
          id: 123,
          username: "user2",
          email: "user2@test.de",
          firstname: "John",
          lastname: "Doee",
          authorities: [],
          isLoggedIn: true
        }
      }
      writeAuthToken("loggedIn")
      resolve(response)
    } else {
      const errorResponse = {
        status: 401
      }
      reject(errorResponse)
    }
  })
}
//axiosInstance.post('/auth/login', qs.stringify({
//    username: username,
//    password: password
//}));

const postCompleteRegistration = (username,
  password,
  firstname = username,
  lastname = username,
  email = username + "@test.com",
  address = "addres1",
  phone = "phoneNr") => {
  return new Promise((resolve, reject) => {
    const errorResponse = {
      status: 401
    }
    reject(errorResponse)
  })
}
//axiosInstance.post('/auth/register', {
//    "username": username,
//    "password": password,
//    "firstname": firstname,
//    "lastname": lastname,
//    "email": email,
//    "address": address,
//    "phone": phone
//});

const postRegistration = (username, firstname, lastname,
  email,
  password) => {
  return new Promise((resolve, reject) => {
    const errorResponse = {
      status: 401
    }
    reject(errorResponse)
  })
}
//axiosInstance.post('/auth/register', {
//    "username": username,
//    "firstname": firstname,
//    "lastname": lastname,
//    "password": password,
//    "email": email,
//});



const auth = {
  postLogin,
  postRegistration,
  postCompleteRegistration
};

const localFirebaseUrl = "http://localhost:5000/pretrendr-d5939/us-central1"
const remoteFirebaseUrl = "https://us-central1-pretrendr-d5939.cloudfunctions.net"
const firebaseUrl = remoteFirebaseUrl
/***** graph *****/
const getWordcountByDay = (query, method, timerange, normalize) => {
  /*return new Promise((resolve, reject)=> {
      if(query.indexOf("trump")>= 0){
          const response = {
              data: trump
          }
          resolve(response)
      }else{
          const errorResponse = {
              status: 404
          }
          reject(errorResponse)
      }
  })*/
  return axios.get(`${firebaseUrl}/getWordcountData?query=${query}&method=${method}`)

  //axiosInstance.get(`/api/gdelt/wordcountByDay/${query}?${timerange}&normalize=${normalize? 'true': 'false'}&method=${method}`);
}
const uploadDataToFirestore = (query, method, timerange, data) => {
  const uploadData = {
    query,
    method,
    timerange,
    data,
  };
  return axios.post(`${firebaseUrl}/addWordcountData`, uploadData)    
}

const getCachedTrends = () => axios.get(`${firebaseUrl}/getAllCachedTerms`)


const graph = {
  getWordcountByDay,
  getCachedTrends
};

const api = {
  auth,
  graph
}

export {
  api
};
