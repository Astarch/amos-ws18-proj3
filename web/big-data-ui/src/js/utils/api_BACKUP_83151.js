import realClient, {api as realApi, logoutUser as realLogoutUser, isLoggedIn as realIsLoggedIn}  from './api.real'
import standaloneClient, {api as standaloneApi, logoutUser as standaloneLogoutUser, isLoggedIn as standaloneIsLoggedIn}  from './api.standalone'
console.dir(standaloneApi)

const isStandaloneClient = false

if(isStandaloneClient){
    //export default standaloneClient;
}else{
    //export default realClient;
}

<<<<<<< HEAD
export const logoutUser = () => {
    writeAuthToken("")
}



/******************** Pretrendr Backend API Endpoints ********************/

/***** auth *****/
const postLogin = (username, password) => axiosInstance.post('/auth/login', qs.stringify({
    username: username,
    password: password
}));

const postCompleteRegistration = (username,
    password,
    firstname = username,
    lastname = username,
    email = username + "@test.com",
    address = "addres1",
    phone = "phoneNr") => axiosInstance.post('/auth/register', {
    "username": username,
    "password": password,
    "firstname": firstname,
    "lastname": lastname,
    "email": email,
    "address": address,
    "phone": phone
});

const postRegistration = (username, firstname,lastname,
    email,
    password) => axiosInstance.post('/auth/register', {
    "username": username,
    "firstname": firstname,
    "lastname": lastname,
    "password": password,
    "email": email,
});

const doLogout = () => axiosInstance.get('/auth/logout');

const auth = {
    postLogin,
    postRegistration,
    postCompleteRegistration,
    doLogout
};

/***** user *****/
const getAll = () => axiosInstance.get('/api/user/getAll');

const user = {
    getAll
};

/***** graph *****/
const getWordcountByDay = (query, method, timerange, normalize) =>
    axiosInstance.get(`/api/gdelt/wordcountByDay/${query}?${timerange}&normalize=${normalize? 'true': 'false'}&method=${method}`);

const graph = {
    getWordcountByDay
};

const uploadDataToFirestore = (query, method, timerange, data) => {
    const uploadData = {
      query,
      method,
      timerange,
      data,
    };
    return axios.post('https://us-central1-pretrendr-d5939.cloudfunctions.net/addWordcountData', uploadData)    
  }

const api = {
    auth,
    user,
    graph,
    uploadDataToFirestore
}
=======
const isLoggedIn = isStandaloneClient? standaloneIsLoggedIn : realIsLoggedIn
const logoutUser = isStandaloneClient? standaloneLogoutUser : realLogoutUser
const api = isStandaloneClient? standaloneApi : realApi
>>>>>>> initial commit to use frontend without backend

export {
    isLoggedIn,
    logoutUser,
    api
}
