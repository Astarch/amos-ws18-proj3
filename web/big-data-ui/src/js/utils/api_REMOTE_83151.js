import realClient, {api as realApi, logoutUser as realLogoutUser, isLoggedIn as realIsLoggedIn}  from './api.real'
import standaloneClient, {api as standaloneApi, logoutUser as standaloneLogoutUser, isLoggedIn as standaloneIsLoggedIn}  from './api.standalone'
console.dir(standaloneApi)

const isStandaloneClient = false

if(isStandaloneClient){
    //export default standaloneClient;
}else{
    //export default realClient;
}

const isLoggedIn = isStandaloneClient? standaloneIsLoggedIn : realIsLoggedIn
const logoutUser = isStandaloneClient? standaloneLogoutUser : realLogoutUser
const api = isStandaloneClient? standaloneApi : realApi

export {
    isLoggedIn,
    logoutUser,
    api
}
