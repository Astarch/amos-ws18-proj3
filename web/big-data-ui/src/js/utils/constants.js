const FormTypeEnum = {"login": 1, "register": 2};
Object.freeze(FormTypeEnum);

export {
  FormTypeEnum
}
