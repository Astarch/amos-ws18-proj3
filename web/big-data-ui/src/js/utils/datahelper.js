function normalizeData(graphdata) {
    //console.time("normalizeData")
    let max = Object.values(graphdata).reduce((max, current) => {
        return current > max ? current : max
    }, -1)
    let normalizeFactor = 1 / (max / 100)
    let normalized = Object.keys(graphdata).reduce((previous, current) => {
        previous[current] = Math.round(graphdata[current] * normalizeFactor)
        return previous;
    }, {});
    //console.timeEnd("normalizeData")
    return normalized
}

function normalizeDataArray(dataArray) {
    let normalizedArray = dataArray.map(value => {
        return Object.assign({}, value, { data: normalizeData(value.data) })
    })
    return normalizedArray
}

function median(values) {
    values = values.slice(0).sort(function(a, b) { return a - b; });

    return middle(values);
}

function middle(values) {
    var len = values.length;
    var half = Math.floor(len / 2);

    if (len % 2)
        return (values[half - 1] + values[half]) / 2.0;
    else
        return values[half];
}

function calculateMinMaxAvgMedianOfData(graphdata) {
    const values = Object.values(graphdata)
    let min = parseInt(Math.min(...values), 10);
    let max = parseInt(Math.max(...values), 10);
    let sum = 0;
    for (var i = 0; i < values.length; i++) {
        sum += parseInt(values[i], 10); //don't forget to add the base
    }
    let avg = parseInt(sum / values.length, 10);
    let _median = parseInt(median(values), 10);

    return Object.assign({}, {
        min,
        max,
        avg,
        median: _median
    })
}

function aggregateDataArray(dataArray) {
    let additionalDataArray = dataArray.map(value => {
        return Object.assign({}, value, calculateMinMaxAvgMedianOfData(value.data))
    })
    return additionalDataArray
}



export {
    normalizeDataArray,
    aggregateDataArray
}