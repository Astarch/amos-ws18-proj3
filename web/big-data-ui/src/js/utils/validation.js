const emailRegex = /^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
const smallCharRegex = /[a-z]+/;
const capitalCharRegex = /[A-Z]+/;
const numberCharRegex = /[0-9]+/;
const specialCharRegex = /[^A-Za-z0-9]+/;


// calculates the strength of a password and returns the strength on a scale from
// 0 to 8 where 8 is a very strong password and 0 a very weak one
const getPasswordStrength = (password) => {
  let strength = 0;

  if (password.length > 0) {
    strength += smallCharRegex.test(password); // contains a-z
    strength += capitalCharRegex.test(password); // contains A-Z
    strength += numberCharRegex.test(password); // contains 0-9
    strength += specialCharRegex.test(password); // contains special character
    strength += Math.floor(Math.min(4, password.length / 3)); // 0-12 characters result in 0-4 points
  }
  return strength;
}

const isMailValid = (mail) => mail.length > 0 && emailRegex.test(mail);
const isPasswordValid = (password) => password.length > 4;
const isNameValid = (name) => name.length > 2;


export {
  isMailValid,
  isPasswordValid,
  isNameValid,
  getPasswordStrength
}

