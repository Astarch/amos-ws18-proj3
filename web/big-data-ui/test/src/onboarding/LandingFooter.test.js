import { shallow, mount } from 'vue-test-utils'
import Vue from 'vue'
import LandingFooter from 'src/js/components/onboarding/LandingFooter'

import {FormTypeEnum} from 'src/js/utils/constants';

describe('LandingFooter.test.js', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(LandingFooter)
    jest.resetModules()
    jest.clearAllMocks()
  })

  test('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

})
