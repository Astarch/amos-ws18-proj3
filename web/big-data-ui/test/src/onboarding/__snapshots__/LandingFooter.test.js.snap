// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingFooter.test.js has the expected html structure 1`] = `
<footer>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-12"
      >
        <ul
          class="list-inline"
        >
          <li>
            <a
              href="#"
            >
              Home
            </a>
          </li>
           
          <li
            class="footer-menu-divider"
          >
            ⋅
          </li>
           
          <li>
            <a
              href="#about"
            >
              About
            </a>
          </li>
           
          <li
            class="footer-menu-divider"
          >
            ⋅
          </li>
           
          <li>
            <a
              href="#services"
            >
              Services
            </a>
          </li>
           
          <li
            class="footer-menu-divider"
          >
            ⋅
          </li>
           
          <li>
            <a
              href="#contact"
            >
              Contact
            </a>
          </li>
        </ul>
         
        <p
          class="copyright text-muted small"
        >
          Copyright © pretrendr 2017. All Rights Reserved
        </p>
      </div>
    </div>
  </div>
</footer>
`;
